src/ServicePulse.Host/app/js/views/pending_retries/controller.js

```diff
@@ -0,0 +1,375 @@
+                var exgroups = data.map(function(obj) {
+                    var nObj = obj;
+                    nObj.panel = 0;
+
+                    return fillRedirect(nObj);
+                });
+
+                vm.pendingRetryMessages = vm.pendingRetryMessages.concat(exgroups);
+                vm.allMessagesLoaded = (vm.pendingRetryMessages.length >= vm.total);
+                vm.page++;
+            }
+            vm.loadingData = false;
+        };
+
+        var init = function() {
+            vm.pendingRetryMessages = [];
+            vm.selectedIds = [];
+            vm.page = 1;
+            vm.filter = {
+                searchPhrase: undefined
+            };
+            vm.filter.start = vm.filter.end = undefined;
+            vm.total = sharedDataService.getstats().number_of_pending_retries;
+            vm.filteredTotal = vm.total;
+            vm.sort = "time_of_failure";
+            vm.direction = "asc";
+            setSortButtonText(vm.sort, vm.direction);
+            vm.loadMoreResults();
+            vm.endpoints = endpointsService.getQueueNames();
+            vm.redirects = redirectService.getRedirects().data;
+        }
+
+        vm.noStatusPresent = function(message) {
+            return (!message.retried || !angular.isDefined(message.retried)) &&
+            (!message.resolved || !angular.isDefined(message.resolved)) && message.number_of_processing_attempts === 1;
+
+        }
+
+        vm.clipComplete = function(messageId) {
+            toastService.showInfo(messageId + ' copied to clipboard');
+        };
+
+        vm.togglePanel = function (message, panelnum) {
+            if (!angular.isDefined(message.messageBody)) {
+                serviceControlService.getMessageBody(message.message_id).then(function (msg) {
+                    message.messageBody = msg.data;
+                }, function () {
+                    message.bodyUnavailable = "message body unavailable";
+                });
+            }
+
+            if (!angular.isDefined(message.messageHeaders)) {
+                serviceControlService.getMessageHeaders(message.message_id).then(function (msg) {
+                    message.messageHeaders = msg.data[0].headers;
+                }, function () {
+                    message.headersUnavailable = "message headers unavailable";
+                });
+            }
+            message.panel = panelnum;
+            return false;
+        };
+
+        vm.toggleRowSelect = function (row) {
+            if (row.retried || row.resolved) {
+                return;
+            }
+
+            row.selected = !row.selected;
+
+            if (row.selected) {
+                vm.selectedIds.push(row.id);
+            } else {
+                vm.selectedIds.splice(vm.selectedIds.indexOf(row.id), 1);
+            }
+        };
+
+        vm.retrySelected = function () {
+            pendingRetryService.retryPendingRetriedMessages(vm.selectedIds).then(function() {
```

Comment:
Should we have a `busy` status here? The `retryPendingRetriedMessages` method could take a while to execute, and the viewmodel properties on the UI are only updated in the success path. Is there a UI indication that the messages are busy being processed?